From 9550f52e11b67dbceb6ae623081ea34a252c7d01 Mon Sep 17 00:00:00 2001
From: Andy CrossGate Yan <GeForce8800Ultra@gmail.com>
Date: Sat, 26 Jul 2025 17:39:28 +0800
Subject: [PATCH 3/3] Revert "BrightnessUtils: Conditionally disable gamma
 conversion for brightness slider"

This reverts commit 745e54af0fb99e8ddcc506aa07afcaa5ff484a88.
---
 .../settingslib/display/BrightnessUtils.java  | 19 -------------------
 1 file changed, 19 deletions(-)

diff --git a/packages/SettingsLib/DisplayUtils/src/com/android/settingslib/display/BrightnessUtils.java b/packages/SettingsLib/DisplayUtils/src/com/android/settingslib/display/BrightnessUtils.java
index 3c06f8ae9972..66ed10433459 100644
--- a/packages/SettingsLib/DisplayUtils/src/com/android/settingslib/display/BrightnessUtils.java
+++ b/packages/SettingsLib/DisplayUtils/src/com/android/settingslib/display/BrightnessUtils.java
@@ -16,15 +16,11 @@
 
 package com.android.settingslib.display;
 
-import android.os.SystemProperties;
 import android.util.MathUtils;
 
 /** Utility methods for calculating the display brightness. */
 public class BrightnessUtils {
 
-    public static final boolean sDisableGammaConversion =
-            SystemProperties.getBoolean("sys.brightness.disable_gamma_conversion", false);
-
     public static final int GAMMA_SPACE_MIN = 0;
     public static final int GAMMA_SPACE_MAX = 65535;
 
@@ -58,11 +54,6 @@ public class BrightnessUtils {
      */
     public static final int convertGammaToLinear(int val, int min, int max) {
         final float normalizedVal = MathUtils.norm(GAMMA_SPACE_MIN, GAMMA_SPACE_MAX, val);
-
-        if (sDisableGammaConversion) {
-            return Math.round(MathUtils.lerp(min, max, normalizedVal));
-        }
-
         final float ret;
         if (normalizedVal <= R) {
             ret = MathUtils.sq(normalizedVal / R);
@@ -86,11 +77,6 @@ public class BrightnessUtils {
      */
     public static final float convertGammaToLinearFloat(int val, float min, float max) {
         final float normalizedVal = MathUtils.norm(GAMMA_SPACE_MIN, GAMMA_SPACE_MAX, val);
-
-        if (sDisableGammaConversion) {
-            return MathUtils.lerp(min, max, normalizedVal);
-        }
-
         final float ret;
         if (normalizedVal <= R) {
             ret = MathUtils.sq(normalizedVal / R);
@@ -142,11 +128,6 @@ public class BrightnessUtils {
      * @return The corresponding slider value
      */
     public static final int convertLinearToGammaFloat(float val, float min, float max) {
-        if (sDisableGammaConversion) {
-            final float normalizedVal = MathUtils.norm(min, max, val);
-            return Math.round(MathUtils.lerp(GAMMA_SPACE_MIN, GAMMA_SPACE_MAX, normalizedVal));
-        }
-
         // For some reason, HLG normalizes to the range [0, 12] rather than [0, 1]
         final float normalizedVal = MathUtils.norm(min, max, val) * 12;
         final float ret;
-- 
2.34.1

